File: languages/python/bloock/entity/authenticity/ens_signer.py
import bloock._bridge.proto.authenticity_entities_pb2 as proto
from bloock.entity.authenticity.signer import Signer
from bloock.entity.authenticity.signer_args import SignerArgs


class EnsSigner(Signer):
    def __init__(self, args: SignerArgs) -> None:
        super().__init__(alg=proto.ENS, args=args)

    def to_proto(self) -> proto.Signer:
        local_key = None
        if self.args.local_key is not None:
            local_key = self.args.local_key.to_proto()

        managed_key = None
        if self.args.managed_key is not None:
            managed_key = self.args.managed_key.to_proto()

        common_name = None
        if self.args.common_name is not None:
            common_name = self.args.common_name.to_proto()

        return proto.Signer(
            alg=self.alg,
            local_key=local_key,
            managed_key=managed_key,
            common_name=common_name
        )

File: languages/php/Bloock/Entity/Authenticity/EnsSigner.php
<?php

namespace Bloock\Entity\Authenticity;

use Bloock\SignerAlg;

class EnsSigner implements Signer
{
    public int $alg;
    public SignerArgs $args;

    public function __construct(SignerArgs $args)
    {
        $this->alg = SignerAlg::ENS;
        $this->args = $args;
    }

    public function toProto(): \Bloock\Signer
    {
        $s = new \Bloock\Signer();
        $s->setAlg($this->alg);

        if ($this->args->localKey != null) {
            $s->setLocalKey($this->args->localKey->toProto());
        }

        if ($this->args->managedKey != null) {
            $s->setManagedKey($this->args->managedKey->toProto());
        }

        if ($this->args->managedCertificate != null) {
            $s->setManagedCertificate($this->args->managedCertificate->toProto());
        }

        if ($this->args->commonName != null) {
            $s->setCommonName($this->args->commonName);
        }

        return $s;
    }
}